lib/app/features/core/permissions/strategies/gallery_permission_strategy.dart

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:async';
+
+import 'package:device_info_plus/device_info_plus.dart';
+import 'package:ice/app/features/core/permissions/strategies/strategies.dart';
+import 'package:permission_handler/permission_handler.dart';
+
+class IosGalleryPermissionStrategy extends BasePermissionStrategy {
+  @override
+  Future<Permission> get permission async => Future.value(Permission.photos);
+}
+
+class AndroidGalleryPermissionStrategy extends BasePermissionStrategy {
```

Comment:
From the perspective of "composition over inheritance" principle, all these permission strategies look redundant. We can pass a permission type to a constructor of the updated `PermissionStrategy` class.

Also, this doesn't look like a strategy. The behaviour stays the same for all the "stragegies", only the type changes, and strategy pattern is about changing behaviour.

Reply:
I created `PermissionHandlerStrategy`, a base class responsible for handling permissions using the `permission_handler` package.